plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchRequestProcessor.java

```diff
@@ -577,8 +621,26 @@ public MLInferenceSearchRequestProcessor create(
             String queryTemplate = ConfigurationUtils.readOptionalStringProperty(TYPE, processorTag, config, QUERY_TEMPLATE);
             Map<String, Object> modelConfigInput = ConfigurationUtils.readOptionalMap(TYPE, processorTag, config, MODEL_CONFIG);
 
-            List<Map<String, String>> inputMaps = ConfigurationUtils.readList(TYPE, processorTag, config, INPUT_MAP);
-            List<Map<String, String>> outputMaps = ConfigurationUtils.readList(TYPE, processorTag, config, OUTPUT_MAP);
+            List<Map<String, String>> inputMaps = ConfigurationUtils.readOptionalList(TYPE, processorTag, config, INPUT_MAP);
+            List<Map<String, String>> outputMaps = ConfigurationUtils.readOptionalList(TYPE, processorTag, config, OUTPUT_MAP);
+
+            List<Map<String, String>> optionalInputMaps = ConfigurationUtils
+                .readOptionalList(TYPE, processorTag, config, OPTIONAL_INPUT_MAP);
+            List<Map<String, String>> optionalOutputMaps = ConfigurationUtils
+                .readOptionalList(TYPE, processorTag, config, OPTIONAL_OUTPUT_MAP);
+
+            if ((inputMaps == null || inputMaps.isEmpty()) && (optionalInputMaps == null || optionalInputMaps.isEmpty())) {
+                throw new IllegalArgumentException(
+                    "Please provide at least one non-empty input_map or optional_input_map for ML Inference Search Request Processor"
+                );
+            }
+
+            if ((outputMaps == null || outputMaps.isEmpty()) && (optionalOutputMaps == null || optionalOutputMaps.isEmpty())) {
```

Comment:
Using `CollectionUtils` could make this condition more readable.

Reply:
added